draft/authentication-oauth2.md

```diff
@@ -0,0 +1,256 @@
+---
+RFC: XXXX
+Author: Kyle Fuller
+Status: Draft
+Created: 2015-09-28
+Last Modified: 2015-09-29
+---
+
+# API Blueprint RFC XXXX: OAuth 2 Authentication Scheme
+
+## Table of Contents
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Rationale](#rationale)
+- [Backwards Compatibility](#backwards-compatibility)
+
+## Abstract
+
+This RFC adds the "OAuth 2" authentication scheme for the Authentication
+framework proposed in RFC XXXX. OAuth 2 is defined in
+[RFC 6749](http://tools.ietf.org/html/rfc6749).
+
+## Motivation
+
+OAuth 2 is a highly common authentication mechanism and it's oftenly used in APIs.
+
+## Rationale
+
+You can use the OAuth 2 Authentication scheme by defining it within the
+"Authentication Schemes" section of an API Blueprint.
+
+```apib
+### Auth (OAuth 2)
+```
+
+### Scopes
+
+Supported scopes may be listed inside the authentication schemes
+
+```apib
+### Auth (OAuth 2)
+
++ scopes
+    + user - A user scope, allowing you access to read the user information
+    + user.email - A user scope, allowing you access to read the user's email addresses
+```
+
+### Endpoints
+
+#### Authorization endpoint
+
+You may describe an authorization endpoint inside the authentication scheme
+using the `authorize` relation identifier.
+
+This endpoint represents the the authorization endpoint described in
+[RFC6749 section 3.1](http://tools.ietf.org/html/rfc6749#section-3.1).
+
+Inside the authorization endpoint you can describe the parameters that this
+endpoint takes along with the parameters it will return during the
+`Redirect Callback`.
+
+The `Redirect` keyword has been added to represent the request that will be
+made to the `Redirect Callback`.
```

Comment:
What? Using the name `Redirect` will be really confusing for people since it doesn't represent the response (which is a redirect) but instead represents the redirect callback.